pkg/microservice/aslan/core/workflow/service/workflow/pipeline_task.go

```diff
@@ -930,15 +934,17 @@ func GetArtifactFileContent(pipelineName string, taskID int64, log *zap.SugaredL
 				return nil, fmt.Errorf("failed to create file %s %v", artifactFileName, err)
 			}
 
-			defer func() {
-				_ = file.Close()
-				_ = os.Remove(path.Join(sourcePath, artifactFileName))
-			}()
-
-			err = s3.Download(context.Background(), s3Storage, artifactFile, file.Name())
+			client, err := s3tool.NewClient(s3Storage.Endpoint, s3Storage.Ak, s3Storage.Sk, s3Storage.Insecure)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create s3 client to download file %s, error: %v", artifactFile, err)
+			}
+			objectKey := s3Storage.GetObjectPath(artifactFile)
+			err = client.Download(s3Storage.Bucket, objectKey, file.Name())
 			if err != nil {
 				return nil, fmt.Errorf("failed to download %s %v", artifactFile, err)
 			}
+			_ = file.Close()
```

Comment:
file won't be closed when an error occured

Reply:
For now I reverted it back to old ways. But it is still a huge memory loss & possible leaks for defer functions to be used in for loop.